llvm/lib/Transforms/InstCombine/InstCombineCalls.cpp

```diff
@@ -1770,6 +1770,14 @@ Instruction *InstCombinerImpl::visitCallInst(CallInst &CI) {
       }
     }
 
+    // smax(smin(abs(x), y), 0) -> smin(abs(x), y) where y >= 0
+    if (IID == Intrinsic::smax &&
+        match(I0, m_SMin(m_c_Intrinsic<Intrinsic::abs>(m_Value(), m_Value()),
```

Comment:
Still need to be `m_c_Intrinsic<..::abs>(m_Value(), m_One())` (And your tests will need toupdated correspondingly.)